packages/app/src/ui/pages/PresentVerifiableCredential/__tests__/PresentVerifiableCredential.test.tsx

```diff
@@ -0,0 +1,144 @@
+
+jest.mock("../usePresentVerifiableCredential", (): unknown => ({
+  ...jest.requireActual("../usePresentVerifiableCredential"),
+  usePresentVerifiableCredential: jest.fn(),
+}));
+
+describe("ui/pages/PresentVerifiableCredential", () => {
+  const mockCryptkeeperVerifiableCredentials = [
+    {
+      verifiableCredential: {
+        context: ["https://www.w3.org/2018/credentials/v1"],
+        id: "http://example.edu/credentials/3732",
+        type: ["VerifiableCredential"],
+        issuer: "did:example:123",
+        issuanceDate: new Date("2020-03-10T04:24:12.164Z"),
+        credentialSubject: {
+          id: "did:example:456",
+          claims: {
+            type: "BachelorDegree",
+            name: "Bachelor of Science and Arts",
+          },
+        },
+      },
+      metadata: {
+        hash: "0x123",
+        name: "Credential #0",
+      },
+    },
+    {
+      verifiableCredential: {
+        context: ["https://www.w3.org/2018/credentials/v1"],
+        id: "http://example.edu/credentials/3733",
+        type: ["VerifiableCredential"],
+        issuer: "did:example:12345",
+        issuanceDate: new Date("2020-03-10T04:24:12.164Z"),
+        credentialSubject: {
+          id: "did:example:123",
+          claims: {
+            type: "BachelorDegree",
+            name: "Bachelor of Science and Arts",
+          },
+        },
+      },
+      metadata: {
+        hash: "0x1234",
+        name: "Credential #1",
+      },
+    },
+  ];
+
+  const mockVerifiablePresentation: IVerifiablePresentation = {
+    context: ["https://www.w3.org/2018/credentials/v1"],
+    type: ["VerifiablePresentation"],
+    verifiableCredential: [
+      {
+        context: ["https://www.w3.org/2018/credentials/v1"],
+        id: "http://example.edu/credentials/3732",
+        type: ["VerifiableCredential"],
+        issuer: "did:example:123",
+        issuanceDate: new Date("2020-03-10T04:24:12.164Z"),
+        credentialSubject: {
+          id: "did:example:456",
+          claims: {
+            type: "BachelorDegree",
+            name: "Bachelor of Science and Arts",
+          },
+        },
+      },
+    ],
+  };
+
+  const defaultHookData: IUsePresentVerifiableCredentialData = {
+    isWalletConnected: true,
+    isWalletInstalled: true,
+    verifiablePresentationRequest: "example presentation request",
+    cryptkeeperVerifiableCredentials: mockCryptkeeperVerifiableCredentials,
+    selectedVerifiableCredentialHashes: ["0x123"],
+    verifiablePresentation: undefined,
+    error: undefined,
+    onCloseModal: jest.fn(),
+    onRejectRequest: jest.fn(),
+    onToggleSelection: jest.fn(),
+    onConfirmSelection: jest.fn(),
+    onReturnToSelection: jest.fn(),
+    onConnectWallet: jest.fn(),
+    onSubmitVerifiablePresentation: jest.fn(),
+  };
+
+  const verifiablePresentationHookData = {
+    ...defaultHookData,
+    verifiablePresentation: mockVerifiablePresentation,
+  };
+
+  beforeEach(() => {
+    createModalRoot();
```

Comment:
Why do you need modal root here?

Reply:
The two components inside, `VerifiableCredentialSelector` and `VerifiableCredentialSigner`, are rendered within a modal so this is needed for finding the DOM elements

Reply:
It shouldn't use Modal as page container. Take a look at CreateIdentity page as an example.

Reply:
Not sure I understand what you mean. The `CreateIdentity` page test uses a `createModalRoot` as well [here](https://github.com/CryptKeeperZK/crypt-keeper-extension/blob/main/packages/app/src/ui/pages/CreateIdentity/__tests__/CreateIdentity.test.tsx#L81). And the `CreateIdentity` component uses the same Modal components.

Reply:
I will remove this `createModalRoot` for `CreateIdentity` tests in my PR because it doesn't use `FullModal` component inside the page component.

Reply:
Got it. Removed it from this change as well.